includes/class-migration.php

```diff
@@ -632,6 +635,57 @@ public static function create_comment_outbox_items( $batch_size = 50, $offset =
 		return null;
 	}
 
+	/**
+	 * Update comment author emails with webfinger addresses for ActivityPub comments.
+	 *
+	 * @param int $batch_size Optional. Number of comments to process per batch. Default 50.
+	 * @param int $offset     Optional. Number of comments to skip. Default 0.
+	 * @return array|null Array with batch size and offset if there are more comments to process, null otherwise.
+	 */
+	public static function update_comment_author_emails( $batch_size = 50, $offset = 0 ) {
+		$comments = \get_comments(
+			array(
+				'number'     => $batch_size,
+				'offset'     => $offset,
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query' => array(
+					array(
+						'key'   => 'protocol',
+						'value' => 'activitypub',
+					),
+				),
```

Comment:
wouldn't it be easier to also check for the empty email field? this way we wouldn't have to care too much about offsets?

Reply:
That's probably good to add. We'd still have to do the meta query for protocol, yes?

Reply:
Looks like [we can't search for empty comment author emails](https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/class-wp-comment-query.php#L757-L759) through `get_comments()`. We'd have to do a direct query.